fmriprep/config.py

```diff
@@ -89,24 +80,23 @@
     # ignoring the most annoying warnings
     import os
     import sys
-    import logging
 
     from uuid import uuid4
     from pathlib import Path
     from time import strftime
-    from niworkflows.utils.spaces import SpatialReferences as _SRs, Reference as _Ref
     from nipype import logging as nlogging, __version__ as _nipype_ver
     from templateflow import __version__ as _tf_ver
     from . import __version__
 
-
-def redirect_warnings(message, category, filename, lineno, file=None, line=None):
-    """Redirect other warnings."""
-    logger = logging.getLogger()
-    logger.debug('Captured warning (%s): %s', category, message)
-
-
-warnings.showwarning = redirect_warnings
+if not hasattr(sys, "_is_pytest_session"):
+    sys._is_pytest_session = False  # Trick to avoid sklearn's FutureWarnings
+# Disable all warnings in main and children processes only on production versions
+if not any((
+    "+" in __version__,
+    __version__.endswith(".dirty"),
+    os.getenv("FMRIPREP_DEV", "0").lower() in ("1", "on", "true", "y", "yes")
+)):
+    os.environ["PYTHONWARNINGS"] = "ignore"
```

Comment:
Circle is not showing the annoying warnings, but it looks like this should be permitting them.

Reply:
from @oesteban mriqc PR:
> If development, warnings.showwarning and warnings.warn are monkeypatched to redirect to the 'py.warnings' logger. The logger writes to /dev/null, but that can we change for debugging purposes.

it might be a good idea to keep these visible in dev versions though, so we can change this.

Reply:
i disabled this monkeypatch on development versions by default, but added a check for `FMRIPREP_WARNINGS` that will enable it.